addons/sale/views/sale_order_line_views.xml

```diff
@@ -25,32 +25,76 @@
         <field name="name">sale.order.line.form.readonly</field>
         <field name="model">sale.order.line</field>
         <field name="arch" type="xml">
-            <form string="Sales Order Item">
+            <form string="Sales Order Item" edit="false">
+                <field name="display_type" invisible="1"/>
                 <sheet>
                     <div class="oe_title">
                         <h1>
-                            <field name="display_name" readonly="1"/>
+                            <field name="display_name"/>
                         </h1>
                     </div>
                     <group>
                         <group>
+                            <field name="product_updatable" invisible="1"/>
                             <field name="order_id" readonly="1"/>
-                            <field name="product_id" readonly="1"/>
-                            <field name="name" readonly="1"/>
-                            <field name="product_uom_qty" readonly="1"/>
-                            <field name="qty_delivered" readonly="1"/>
-                            <field name="qty_invoiced"/>
-                            <field name="product_uom" readonly="1"/>
+                            <field name="product_uom" invisible="1"/>
+                            <field name="product_uom_category_id" invisible="1"/>
+                            <field name="product_uom_readonly" invisible="1"/>
+                            <field name="qty_delivered_method" invisible="1"/>
+                            <field name="state" invisible="1"/>
+                            <field name="product_id"
+                                context="{
+                                    'partner_id': order_partner_id,
+                                    'quantity': product_uom_qty,
+                                    'uom': product_uom,
+                                    'company_id': company_id,
+                                }"
+                                attrs="{
+                                    'readonly': [
+                                        '&amp;',
+                                            ('id', '!=', False),
+                                            ('product_updatable', '=', False),
+                                    ],
+                                    'required': [('display_type', '=', False)],
+                                }"
+                                force_save="1"
+                                widget="many2one_barcode"
+                            />
+                            <field name="name"/>
+                            <field name="product_uom_qty" context="{'partner_id': order_partner_id, 'quantity': product_uom_qty, 'uom': product_uom, 'uom_qty_change': True, 'company_id': company_id}"/>
```

Comment:
Same here, I don't believe this context adds anything at all, there is no specific widget, and the field is not relational, so it won't trigger any subsequent action where this context could be useful.